File: lib/models/action_plan.dart
class ActionPlan {
  final String id;
  final String selectedAction;
  final String reasoning;
  final Map<String, dynamic> parameters;
  final List<Map<String, dynamic>> fallbackActions;

  ActionPlan({
    required this.id,
    required this.selectedAction,
    required this.reasoning,
    required this.parameters,
    required this.fallbackActions,
  });

  ActionPlan.fromJson(Map<String, dynamic> j)
      : id = j['id'],
        selectedAction = j['selected_action'],
        reasoning = j['reasoning'],
        parameters = j['parameters'],
        fallbackActions = List<Map<String, dynamic>>.from(j['fallback_actions']);
}

File: backend/models/action_plan.py
from pydantic import BaseModel
from typing import Optional, List, Any, Dict


class FallbackAction(BaseModel):
    action: str
    trigger: str


class ActionPlan(BaseModel):
    id: Optional[str] = None
    insight_id: Optional[str] = None
    selected_action: str
    reasoning: str
    parameters: Dict[str, Any] = {}
    fallback_actions: List[Any] = []
